addons/mrp/tests/test_bom.py

```diff
@@ -845,13 +845,20 @@ def test_22_bom_report_recursive_bom(self):
             'uom_po_id': uom_unit.id,
         })
 
-        semi_finished = self.env['product.product'].create({
+        semi_finished_1 = self.env['product.product'].create({
```

Comment:
No need to create an additional product, the test is already incorrect without it. We have:
```
80 Finished, so we need:
- 4 Semi-Finished (BoM: for 100 Finished, we need 5 Semi-Finished), which means:
  - 0.73 Assembly (BoM: for 11 Semi Finished, we need 2 Assembly), which means:
    - 0.58 Raw Material (BoM: for 5 Assembly, we need 4 Raw Material)
```
And, since one raw material costs $5, 80 Finished costs $2.92 (and not $0.58 as it was expected by the current test). I guess this test was edited a little too quickly by d113cb1e32240386b3363698cd650d43f6c351db

Therefore, you can remove the test modifications and just correct the `assertAlmostEqual` (that way, we don't make the flow more complex and the test will match again its description)